edward/util/metrics.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+from functools import wraps
+
+import tensorflow as tf
+
+
+def with_binary_averaging(metric):
```

Comment:
Great idea to include this. Can you explain how the function works as a wrapper? That is, how does a user specify the average option in `ed.evaluate`?

Reply:
Ah. Still need to add this. I'll add a test for it as well.

Thinking `ed.evaluate(['mean_absolute_error', ('mean_squared_error', {'average': 'micro'})], {x: x_data}, n_samples=1)`, for example.

Reply:
Oof. It sounds difficult to specify intuitively. I like your approach for now; let's leave it as the default average option.

Reply:
Working on this one now.

Reply:
Added.

Reply:
Added.